sdk/core/azure-json/src/main/java/com/azure/json/JsonBoolean.java

```diff
@@ -0,0 +1,132 @@
+package com.azure.json;
+
+import java.io.IOException;
+
+/**
+ * Class representing the JSON boolean type
+ */
+public final class JsonBoolean extends JsonElement {
+    /**
+     * Stores the String representation of the JsonBoolean object.
+     * Can only be either "true" or "false".
+     */
+    private boolean booleanValue;
+
+    /**
+     * Default constructor
+     * Default sets booleanValue to "true" through the other constructor.
+     *
+     * TODO: may need to remove this due to design guidelines? May only want to
+     * have the public JsonBoolean(boolean value) constructor.
+     */
+
+    /**
+     * Constructor used to set the value of the JsonBoolean.
+     *
+     * @param value the boolean value to set the JsonBoolean object to. Either
+     * true or false. If value is true, then booleanValue set to "true"; otherwise,
+     * set to "false"
+     */
+    private JsonBoolean(boolean value) { 
+        this.booleanValue = value; 
+    }
+
+    /**
+     * Lazy-loading the Singleton instances of JsonBoolean of TRUE & FALSE.
+     */
+    private static class LoadSingleton {
+        static final JsonBoolean TRUE = new JsonBoolean(true);
+        static final JsonBoolean FALSE = new JsonBoolean(false);
+    }
```

Comment:
These objects are very cheap, I don't feel we need to add this overhead to defer their instantiation.

Reply:
@alzimmermsft this should now be done